### PR TITLE
ImageStreamImage returns incorrect image info

### LINKED ISSUE #6919: UI not reflecting correct deployment
deployment # 4 
build  # 1 
screen shot 2016-01-30 at 12 57 12 pm

This also happens on the deployments tab.

### PR DESCRIPTION
**Pull request by smarterclayton**
Go range variable can't be the value we take a reference to.

Fixes #6919

@brenton this is a 3.1.1.1 candidate fix as well as 1.1.1.2

### CODE REVIEW

**Comment by smarterclayton** [comment]
@miminar @soltysh review please

File: pkg/image/api/helper.go [hunk 1]
```diff
@@ -635,16 +635,17 @@ func ResolveImageID(stream *ImageStream, imageID string) (*TagEvent, error) {
 	var event *TagEvent
 	set := sets.NewString()
 	for _, history := range stream.Status.Tags {
```
Comment (liggitt) [inline pkg/image/api/helper.go]: do you need to do the same thing to avoid sharing `history`?

Comment (smarterclayton) [inline pkg/image/api/helper.go]: History has a pointer to a slice of items, so no.

**Comment by pweil-** [comment]
LGTM

**Comment by smarterclayton** [comment]
[merge]

**Comment by smarterclayton** [comment]
[test]

**Comment by smarterclayton** [comment]
[test]

**Pull request merged**